pyvista/core/filters/image_data.py

```diff
@@ -1331,17 +1504,28 @@ def pad_image(
             - Six values, one for each ``(-X, +X, -Y, +Y, -Z, +Z)`` boundary, to apply
               padding to each boundary independently.
 
+        dimensionality : bool, VectorLike[bool], Literal['0D', '1D', '2D', '3D', 'preserve'], default: 'preserve'
+            Control which dimensions will be modified by the filter.
+
+            - ``True``: all dimensions are modified.
+            - ``False``: no dimensions are modified.
+            - Can be specified as a sequence of 3 boolean allow modification on a per
+                dimension basis.
+            - ``'0D'``: convenience alias to output a 0D ImageData, if possible, i.e.,
+                dims will be similar to (1, 1, 1).
+            - ``'1D'``: convenience alias to output a 1D ImageData, if possible, i.e.,
+                dims will be similar to (>1, 1, 1).
+            - ``'2D'``: convenience alias to output a 2D ImageData, if possible, i.e.,
+                dims will be similar to (>1, >1, 1).
+            - ``'3D'``: convenience alias to output a 3D ImageData, if possible, i.e.,
+                dims will be similar to (>1, >1, >1).
+            - ``'preserve'``: convenience alias to not modify singleton dimensions.
```

Comment:
Suggest removing `True` and `False` options as suggested earlier, and using padding-specific language. Also, the `0D` case is only valid here if the filter does nothing (i.e. no padding is applied). Is it worth it to include this case in the docs? 

```suggestion
            Control which dimensions will be padded by the filter.

            - Can be specified as a sequence of 3 boolean apply padding on a per
                dimension basis.
            - ``'1D'``: apply padding such that the output is a 1D ImageData where exactly one dimension is greater than one, e.g. ``(>1, 1, 1)``. Only valid for 0D or 1D inputs.
            - ``'2D'``: apply padding such that the output is a 2D ImageData where exactly two dimensions are greater than one, e.g. ``(>1, >1, 1)``. Only valid for 0D, 1D, or 2D inputs.
            - ``'3D'``: apply padding such that the output is a 3D ImageData, where all three dimensions are greater than one, e.g. ``(>1, >1, >1)``. Valid for any 0D, 1D, 2D, or 3D inputs.
```